lib/streamlit/navigation/page.py

```diff
@@ -296,7 +297,8 @@ def run(self) -> None:
                 self._page()
                 return
             else:
-                code = ctx.pages_manager.get_page_script_byte_code(str(self._page))
+                runtime = get_instance()
+                code = runtime.get_page_script_byte_code(str(self._page))
```

Comment:
This seems like it makes sense to just grab this from the runtime.

Reply:
When we execute the `run` function, I believe we must have a script runner. So far, we seem to pass the `_script_cache` from the runtime to the `AppSession` and then to the `ScriptRunner`, where a similar line of code is executed as in the here newly introduced `runtime.get_page_script_byte_code`: `code = self._script_cache.get_bytecode(script_path)` ([here](https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/runtime/scriptrunner/script_runner.py#L502C56-L502C67)). 
Hence I am personally leaning towards keeping it in the runner or context.

Reply:
How would the page get access to the ScriptRunner with the ScriptCache? I believe the only way is through the context. We can add the ScriptCache to the context, but that seems like overkill. I perhaps would argue that we do not pass down the script cache cause it's essentially a file -> bytecode cache which is unique to the app (or runtime) and really is not special to any particular session/scriptrunner.

Reply:
Good point about the ScriptRunner. However, I think I liked the previous design better where the `script_cache` was passed to the pages_manager instead of a new function created in runtime that accesses the script_cache globally. The reason for this is because we pass the `script_cache` already down the chain, so it feels a little bit mixed up that sometimes we use the passed down object and now here the global one. So, I personally would stick to the existing pattern. But I don't feel super strong about it, but it smells like we should add it to the tech debt list and look into cleaning it up at some point. Unless my assessment is totally wrong and it makes perfect sense, then don't - its more of a gut feeling than a deep analysis 😁